midx.c

```diff
@@ -1040,15 +1062,32 @@ int verify_midx_file(const char *object_dir)
 	}
 
 	progress = start_progress(_("Verifying object offsets"), m->num_objects);
+
+	/*
+	 * Create an array mapping each object to its packfile id.  Sort it
+	 * to group the objects by packfile.  Use this ordering to visit each
+	 * of the objects, but only require 1 packfile to be open at a time.
+	 */
+	ALLOC_ARRAY(pairs, m->num_objects);
```

Comment:
For very large repos, this could be asking for many GB, right? Should a failure to alloc cause a failure in midx validation?

Reply:
Jeff and I talked offline and I misunderstood the number of object at play. We should ask for less than 500MB, which should be serviceable.